pkg/gitparse/gitparse.go

```diff
@@ -363,6 +372,12 @@ func (c *Parser) FromReader(ctx context.Context, stdOut io.Reader, diffChan chan
 	}
 	currentDiff := diff(currentCommit)
 
+	createEmptyDiffForCommit := func(commit *Commit) *Diff {
+		// Initialize an empty Diff instance associated with the given commit.
+		// Since this diff represents "no changes", we only need to set the commit.
+		return &Diff{Commit: commit}
+	}
+
```

Comment:
I'm usually all for functional hooliganism but this seems like overkill even to me. It's a one-line function used in one place - why not just inline it?